api/test/media_integration.py

```diff
@@ -127,6 +129,22 @@ def thumb_webp(fixture):
     assert thumbnail_response.headers["Content-Type"] == "image/webp"
 
 
+def thumb_full_size(fixture):
+    def _get_image_dimen(url: str) -> tuple[int, int]:
+        response = requests.get(url)
+        image = Image.open(BytesIO(response.content))
+        return image.size
+
+    thumbnail_url = fixture["results"][0]["thumbnail"]
+    full_w, full_h = _get_image_dimen(f"{thumbnail_url}?full_size=yes")
+    scaled_w, scaled_h = _get_image_dimen(thumbnail_url)
+    if full_w > 600:
+        assert scaled_w == 600
+        assert full_w > scaled_w
+    else:
+        assert scaled_w == full_w  # h2non/imaginary will not scale up
```

Comment:
Hmm... This seems like solving the flakiness by hiding it rather than actually making the test _do_ the testing every time though.

Should we choose specific fixture data that we know will test the full size operation correctly? We wouldn't write a unit test using data we know won't actually exercise the unit we want to test, after all. Are integration tests different?

Reply:
It does at least test that the image returned using `full_screen` is either larger (when the source material > 600px) or the same size (when the source material is 600px or lower).